easybuild/easyblocks/generic/systemmpi.py

```diff
@@ -0,0 +1,192 @@
+
+    def __init__(self, *args, **kwargs):
+        """Extra initialization: determine system MPI version, prefix and any associated envvars."""
+        super(SystemMPI, self).__init__(*args, **kwargs)
+
+        # Determine MPI path (real path, with resolved symlinks) to ensure it exists
+        mpi_name = self.cfg['name'].lower()
+
+        # Check we have MPI wrappers
+        if mpi_name != 'impi':
+            mpi_c_wrapper = 'mpicc'
+        else:
+            mpi_c_wrapper = 'mpiicc'
+        path_to_mpi_c_wrapper = which(mpi_c_wrapper)
+        if path_to_mpi_c_wrapper:
+            path_to_mpi_c_wrapper = os.path.realpath(path_to_mpi_c_wrapper)
+            self.log.info("Found path to MPI implementation '%s' %s compiler (with symlinks resolved): %s",
+                          mpi_name, mpi_c_wrapper, path_to_mpi_c_wrapper)
+        else:
+            raise EasyBuildError("%s not found in $PATH", mpi_c_wrapper)
+
+        # Determine compiler version and installation prefix
+        if mpi_name == 'openmpi':
+            output_of_ompi_info, _ = run_cmd("ompi_info", simple=False)
+            # Extract the version of OpenMPI
+            self.mpi_version = self.extract_ompi_setting("Open MPI", output_of_ompi_info)
+
+            # Extract the installation prefix
+            self.mpi_prefix = self.extract_ompi_setting("Prefix", output_of_ompi_info)
+
+            # Extract any OpenMPI environment variables in the current environment and ensure they are added to the
+            # final module
+            raw_env = os.environ
+            self.mpi_envvars = dict((key, value) for key, value in raw_env.iteritems() if key.startswith("OMPI_"))
+
+            # Extract the C compiler used underneath OpenMPI, check for the definition of OMPI_MPICC
+            self.mpi_c_compiler = self.extract_ompi_setting("C compiler", output_of_ompi_info)
+
+        # TODO :Add support for impi
+        else:
+            raise EasyBuildError("Unknown system MPI implementation %s", mpi_name)
+
+        self.log.debug("Derived version/install prefix for system MPI %s: %s, %s",
+                       mpi_name, self.mpi_version, self.mpi_prefix)
+
+        # For the version of the underlying C compiler need to explicitly extract (to be certain)
+        self.c_compiler_version, debug_msg = extract_compiler_version(self.mpi_c_compiler)
+        self.log.debug(debug_msg)
+        self.log.debug("Derived compiler/version for C compiler underneath system MPI %s: %s, %s",
+                       mpi_name, self.mpi_c_compiler, self.c_compiler_version)
+
+        # If EasyConfig specified "real" version (not 'system' which means 'derive automatically'), check it
+        if self.cfg['version'] == 'system':
+            self.log.info("Found specified version '%s', going with derived MPI version '%s'",
+                          self.cfg['version'], self.mpi_version)
+        elif self.cfg['version'] != self.mpi_version:
```

Comment:
use positive logic, use `log.info` to inform that specified version matches found version